Builds fail in docker when `USE_PCH=OFF`

# Bug reports:

### Expected behavior:

After initializing a build directory with
```
cmake -D CHARM_ROOT=/work/charm/multicore-linux64-clang/ -D CMAKE_CXX_COMPILER=/usr/bin/clang++-5.0 -D CMAKE_Fortran_COMPILER=/usr/bin/gfortran-7 -D USE_PCH=OFF /path/to/spectre/
```
in the docker container sxscollaboration/spectrebuildenv:latest, we should expect to be able to build `bin/RunTests` (or indeed, the other executables) with `make` commands.

Omitting the `USE_PCH=OFF` flag gives expected behavior.

### Current behavior:

With the `USE_PCH=OFF` flag, linker errors are emitted regarding nodelocks from charm++. Examples:
```
/home/mox/research/SXS/spectre/src/Parallel/NodeLock.hpp:26: error: undefined reference to 'LrtsDestroyLock(void*)'
/home/mox/research/SXS/spectre/src/Parallel/NodeLock.hpp:37: error: undefined reference to 'LrtsLock(void*)'
/home/mox/research/SXS/spectre/src/Parallel/NodeLock.hpp:65: error: undefined reference to 'LrtsUnlock(void*)'
```

### Environment:

```
CMake Version: 3.10.2
SpECTRE Version: 0.0.0
SpECTRE branch: cce_equations
SpECTRE hash: 5b7452777fd6c1b67ab4948d61184bdfb8039e29
Hostname: 00d1d48f99f8
Host system: Linux-4.17.1-gentoo
Host system version: 4.17.1-gentoo
Host system processor: x86_64
CMAKE_CXX_FLAGS:  -fPIC
CMAKE_CXX_LINK_FLAGS: 
CMAKE_CXX_FLAGS_DEBUG: -g
CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_C_FLAGS:  -fPIC
CMAKE_C_LINK_FLAGS: 
CMAKE_C_FLAGS_DEBUG: -g
CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_Fortran_FLAGS: 
CMAKE_Fortran_LINK_FLAGS: 
CMAKE_Fortran_FLAGS_DEBUG: -g
CMAKE_Fortran_FLAGS_RELEASE: -O3 -DNDEBUG -O3
CMAKE_EXE_LINKER_FLAGS: 
CMAKE_BUILD_TYPE: Debug
CMAKE_CXX_COMPILER: /usr/bin/clang++-5.0
CMAKE_CXX_COMPILER_VERSION: 5.0.1
CMAKE_C_COMPILER: /usr/bin/cc
CMAKE_C_COMPILER_VERSION: 7.3.0
CMAKE_Fortran_COMPILER: /usr/bin/gfortran-7
CMAKE_Fortran_COMPILER_VERSION: 7.3.0
Python Version: 2.7.15
Charm Version:  6.8.0
Charm SMP:  YES
BLAS_LIBRARIES:  /usr/lib/x86_64-linux-gnu/libopenblas.so
Blaze Version:  3.2
Boost Version:  1.65.1
Catch Version:  2.1.0
Google Benchmark Found
GSL Version:  2.5
HDF5 Version:  1.10.0.1
jemalloc Version:  ..
LIBXSMM Version:  ..
LAPACK_LIBRARIES:  /usr/lib/x86_64-linux-gnu/libopenblas.so;/usr/lib/x86_64-linux-gnu/libopenblas.so
NumPy Version:  1.16.0
SciPy Version:  1.2.0
```
### Discussion
Particularly mysterious is that the apparent suspect, charm++ libraries, do not seem to have anything to do with the SpECTRE pre-compiled header, which should be the only thing that is substantively changed by the cmake `USE_PCH` flag.

--- a/src/IO/Observer/ReductionActions.hpp
+++ b/src/IO/Observer/ReductionActions.hpp
@@ -31,6 +31,12 @@
 #include "Utilities/TaggedTuple.hpp"
 
 namespace observers {
+
+/// \cond
+template <class Metavariables>
+struct ObserverWriter;
+/// \endcond
+
 namespace ThreadedActions {
 /// \cond
 struct WriteReductionData;

--- a/src/IO/Observer/Tags.hpp
+++ b/src/IO/Observer/Tags.hpp
@@ -3,8 +3,8 @@
 
 #pragma once
 
+#include <converse.h>
 #include <cstddef>
-#include <lrtslock.h>
 #include <string>
 #include <unordered_map>
 #include <unordered_set>

--- a/src/Parallel/Algorithm.hpp
+++ b/src/Parallel/Algorithm.hpp
@@ -4,10 +4,10 @@
 #pragma once
 
 #include <boost/variant/variant.hpp>
+#include <converse.h>
 #include <cstddef>
 #include <exception>
 #include <initializer_list>
-#include <lrtslock.h>
 #include <ostream>
 #include <pup.h>
 #include <tuple>

--- a/src/Parallel/NodeLock.hpp
+++ b/src/Parallel/NodeLock.hpp
@@ -3,7 +3,7 @@
 
 #pragma once
 
-#include <lrtslock.h>
+#include <converse.h>
 
 #include "Utilities/Gsl.hpp"
 #include "Utilities/NoSuchType.hpp"

--- a/tests/Unit/ActionTesting.hpp
+++ b/tests/Unit/ActionTesting.hpp
@@ -10,10 +10,10 @@
 #include <boost/preprocessor/logical/not.hpp>
 #include <boost/preprocessor/punctuation/comma_if.hpp>
 #include <boost/preprocessor/repetition/repeat.hpp>
+#include <converse.h>
 #include <cstddef>
 #include <deque>
 #include <exception>
-#include <lrtslock.h>
 #include <memory>
 #include <ostream>
 #include <random>

--- a/tests/Unit/NumericalAlgorithms/LinearSolver/ResidualMonitorActionsTestHelpers.hpp
+++ b/tests/Unit/NumericalAlgorithms/LinearSolver/ResidualMonitorActionsTestHelpers.hpp
@@ -4,13 +4,17 @@
 #pragma once
 
 #include <cmath>
+#include <converse.h>
 #include <string>
 
 #include "DataStructures/DataBox/DataBox.hpp"
 #include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ReductionActions.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/ActionTesting.hpp"
 
 namespace Parallel {
 template <typename Metavariables>

